src/tree.h

```diff
@@ -12,17 +12,20 @@
 #include "odb.h"
 #include "vector.h"
 #include "strmap.h"
+#include "pool.h"
 
 struct git_tree_entry {
 	uint16_t attr;
 	git_oid oid;
+	bool pooled;
```

Comment:
It does indeed bring sadness to alignment, I forgot to double-check after testing it out. It's probably enough to have a `uint16_t` as length for anything we would actually care to support, tbh (and lol if you actually need the 64 bits for your path length).